Mathlib/RingTheory/PowerSeries/Restricted.lean

```diff
@@ -0,0 +1,147 @@
+/-
+Copyright (c) 2025 William Coram. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: William Coram
+-/
+
+import Mathlib.Algebra.Group.NatPowAssoc
+import Mathlib.Analysis.Normed.Group.Ultra
+import Mathlib.Analysis.RCLike.Basic
+import Mathlib.RingTheory.PowerSeries.Basic
+
+/-!
+# Restricted power series
+
+We say a powerseries over a normed ring `R` is restricted for a parameter `c` if
+`‖coeff R i f‖ * c^i → 0`.
+
+-/
+
+namespace PowerSeries
+
+open PowerSeries Filter
+open scoped Topology
+
+/-- A power series over `R` is restricted of paramerter `c` if we have the following limit. -/
+def IsRestricted {R : Type*} [NormedRing R] (c : ℝ) (f : PowerSeries R) :=
+  Tendsto (fun (i : ℕ) => (norm (coeff R i f)) * c^i) atTop (𝓝 0)
+
+namespace Restricted
+
+variable {R : Type*} [NormedRing R] (c : ℝ)
+
+lemma isRestricted_iff (f : PowerSeries R) : IsRestricted c f ↔ IsRestricted |c| f := by
+  simp_rw [IsRestricted, NormedAddCommGroup.tendsto_atTop, sub_zero, norm_mul, norm_norm, norm_pow,
+    Real.norm_eq_abs, abs_abs]
+
+lemma zero : IsRestricted c (0 : PowerSeries R) := by
+  simp_rw [IsRestricted, map_zero, norm_zero, zero_mul, tendsto_const_nhds_iff]
+
+lemma one : IsRestricted c (1 : PowerSeries R) := by
```

Comment:
Using my suggestion to have `isRestricted_iff` this can become
```lean
lemma one : IsRestricted c (1 : PowerSeries R) := by
  simp only [isRestricted_iff, coeff_one, norm_mul, norm_pow, Real.norm_eq_abs]
  refine fun _ _ ↦ ⟨1, fun n hn => ?_ ⟩
  split
  · omega
  · simpa
```